Refactor Presets and Supporting Documentation

indiegenus opened the pull request:
# Notes

- updated all Presets and their respective pages in Docs
- cleaned up and reorganized the sidebar menu for Docs
  - removed "Existing Project" page
  - removed Presets "Introduction" page
  - moved "Installation" and "How It Works" pages up to top-level with "Introduction" and "Demo"
  - renamed "Adonis" preset page to "AdonisJS"
  - renamed "Wordpress" preset page to "WordPress" (with uppercase "P")
  - added a new section called "Snippets", with first snippet for adding "MailHog" to a local dev environment
- WordPress preset:
  - added a `.env` file to the project to make `kool run mysql` work
  - changed DB_DATABASE to match WP default `wordpress` (instead of `database`); via the new `.env` file
- Hugo preset:
  - added a `quickstart` script in the `kool.yml` to facilitate the creation of a Hello World experience
- added new basic, Hello World presets for PHP and NodeJS
  - for now, at least, I decided to keep the "Starting New Project" page as is, and **not** use the new PHP preset, because it better explains how to add Kool to **any** project, when a preset is not available.

# TODO

- [ ] fix permissions issues with Hugo and WordPress (see https://github.com/kool-dev/kool/issues/316)
- [ ] fix the absence of commented out lines in the auto-generated `kool.yml` files (see https://github.com/kool-dev/kool/issues/312#issuecomment-830670212)
- [ ] move all Presets out of 2-Presets into 1-Getting Started and deprecate 2-Presets as a section in the sidebar menu
- [ ] finish testing all variants (based on wizard selections)

| Issue | # |
| -----: | :-----: |
| :pencil: Refactor | Yes |
| :open_book: Documentation | Yes |

fabriciojs commented on docs/2-Presets/WordPress.md:
```diff
@@ -0,0 +1,234 @@
+> npm
+  yarn
+
+? What composer version do you want to use [Use arrows to move, type to filter]
```
Indeed we seemed to already have Composer inquiry for creating Wordpress app... does that even make sense? Should we remove?

indiegenus replied:
@fabriciojs - I think you're right, that it's not applicable to WP and we should remove. Awaiting agreement from @dbpolito

dbpolito replied:
yup, we can remove composer from wordpress...

fabriciojs commented on docs/5-Snippets/Mailhog.md:
```diff
@@ -0,0 +1,102 @@
+# Mailhog
```
Just a comment - we have very common snippets for `elasticsearch`, `ssh tunneling`, `caddy proxy` - which we can add next to have a better body of content in this section to start with.

fabriciojs reviewed (approved):
Great job @indiegenus 👍🏻

indiegenus commented:
@dbpolito @fabriciojs - see https://github.com/kool-dev/kool/pull/317/commits/d86b005fb951625bd13f96a1f59646f25d90489f with updates to PR based on your feedback:
- removed Composer from WordPress preset
- added permissions-related comments to Hugo and WordPress docs (for Linux and Windows WSL devs)
